src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -212,12 +212,16 @@ private static FilePath tempDir(FilePath ws) {
         private final String[] envHost;
         private final String ws;
         private final @CheckForNull String toolName;
+        private final boolean hasEnv;
+        private final boolean hasWorkdir;
 
-        Decorator(String container, EnvVars envHost, String ws, String toolName) {
+        Decorator(String container, EnvVars envHost, String ws, String toolName, VersionNumber dockerVersion) {
             this.container = container;
             this.envHost = Util.mapToEnv(envHost);
             this.ws = ws;
             this.toolName = toolName;
+            this.hasEnv = dockerVersion.isOlderThan(new VersionNumber("1.13.0"));
```

Comment:
Wait, we have `--env` up to 1.12.x, but this was removed as of 1.13.0? Did you forget a `!`?

BTW I find `!a.isOlderThan(b)` a bit confusing. `a.compareTo(b) >= 0` may be clearer.

Reply:
Indeed missing negation here. 
VersionNumber unfortunately don't have an `NewerOrEquals()` method, so this negation. We can indeed use a plain `compareTo`.